bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/dialogs/SearchPattern.java

```diff
@@ -39,21 +39,30 @@ public class SearchPattern {
 
 	/**
 	 * Match rule: The search pattern is a prefix of the search result.
+	 * <p>
+	 * This is the <em>default</em> match rule if no other applies.
```

Comment:
I don't think this should be documented here, but more on the method/constructor that do take this decision.

Reply:
OK, I think I get the point here. At one of the constructors, it is de-facto mentioned?:

```java
        /** ...
	 *                     Note also that default behavior for generic types/methods
	 *                     search is to find prefix matches.
	 */
	public SearchPattern(int allowedRules) {
		this.allowedRules = allowedRules;
		this.substringSearch = (allowedRules & RULE_SUBSTRING_MATCH) != 0;
	}
```

I'm not sure what exactly this sentence means by "generic types/methods" though, it is pretty old code already. So what do you suggest? Maybe to move this info just to the `matches()` method, to keep it simple, less confusing and just at one place?

Reply:
So I've moved all this stuff to the `matches()` method, I hope it makes things better.

Maybe a generally better approach would be to move much of the scattered documentation to the javadoc of the class itself. But that would be a major overhaul which I haven't originally planned for this ticket... :)

Reply:
Seems like accepted, closing thread.